kasa/smart/smartdevice.py

```diff
@@ -486,3 +486,26 @@ async def factory_reset(self) -> None:
         Note, this does not downgrade the firmware.
         """
         await self.protocol.query("device_reset")
+
+    @property
+    def device_type(self) -> DeviceType:
+        """Return the device type."""
+        if self._device_type is not DeviceType.Unknown:
+            return self._device_type
+
+        if self.children:
+            if "SMART.TAPOHUB" in self.sys_info["type"]:
+                pass  # TODO: placeholder for future hub PR
+            else:
+                self._device_type = DeviceType.Strip
+        elif "light_strip" in self._components:
+            self._device_type = DeviceType.LightStrip
+        elif "dimmer_calibration" in self._components:
+            self._device_type = DeviceType.Dimmer
+        elif "brightness" in self._components:
+            self._device_type = DeviceType.Bulb
+        else:
```

Comment:
I might be missing it but I can't see where we would set this to `Plug` without a warning.
```suggestion
	elif "PLUG" in self.sys_info["type"]:
		self._device_type = DeviceType.Plug
        else:
```

Reply:
Ah, good catch :+1: